pykanidm/kanidm/types.py

```diff
@@ -4,14 +4,19 @@
 
 from ipaddress import IPv4Address, IPv6Address, IPv6Network, IPv4Network
 import socket
-from typing import Any, Dict, List, Optional
+from typing import Any, Dict, List, Optional, Generic, TypeVar
 from urllib.parse import urlparse
 
-from pydantic import field_validator, ConfigDict, BaseModel, Field, RootModel
+from pydantic import field_validator, ConfigDict, BaseModel, Field
 import toml
 
+from kanidm.models.group import Group as GroupInfo, RawGroup as RawGroupInfo, GroupList  # noqa
 
-class ClientResponse(BaseModel):
+
+T = TypeVar("T", Dict[str, Any], None)
```

Comment:
This is not just a Dict[str, Any] = it can also be a string / list / etc - it's JSON-decoded from the payload.